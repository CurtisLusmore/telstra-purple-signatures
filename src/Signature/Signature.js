import React from 'react';

import SocialMedia from '../SocialMedia';

const Signature = props => {
  const {
    brandLogo,
    name,
    placeholders,
    brandName,
    title,
    isSupport,
    qualifications,
    mobileHtml,
    supportMobile,
    emailHtml,
    supportEmail,
    twitterHtml,
    brandLink,
    brandLinkName,
  } = props;
  return <div>
    <table border="0" cellSpacing="0" cellPadding="0" width="100%"
           style={{
             color: '#3D5567',
             fontFamily: 'Segoe UI, Tahoma, Geneva, Verdana, sans-serif',
             fontSize: '10.5pt'
           }}>
      <tr>
        <td valign="top" style={{ borderRight: '1px solid #3D5567', paddingRight: '20px', width: '76px' }}>
          <img
            width="73" height="53"
            src={brandLogo.link}
            alt={brandLogo.alt}/>
        </td>
        <td valign="top" style={{ paddingLeft: '20px' }}>
          <p style={{ marginBottom: '10px' }}>
            <b>{name || placeholders.name}</b>
            <br/>
            {brandName}&nbsp;|&nbsp;{title || placeholders.title}
          </p>
          {qualifications ? <p style={{ marginBottom: '10px' }}>{qualifications}</p> : null}
          <p style={{ marginBottom: '20px' }}>
                <span>
                  <b>M</b>
                  &nbsp;{mobileHtml}
                  &nbsp;{supportMobile}
                </span>
            {isSupport ? null : <span>&nbsp;&nbsp;</span>}
            <span>
                  <b>E</b>
              &nbsp;{emailHtml}
              &nbsp;{supportEmail}
                </span>
            {twitterHtml}
            <span>
                  <b>W</b>&nbsp;
              <a href={brandLink}>{brandLinkName}</a>
                </span>
          </p>
          <SocialMedia/>
        </td>
      </tr>
    </table>
  </div>;
};
export default Signature