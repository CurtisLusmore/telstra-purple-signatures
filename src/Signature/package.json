{
  "main": "./Signature.js",
  "name": "Signature",
  "version": "0.0.0",
  "private": true
}