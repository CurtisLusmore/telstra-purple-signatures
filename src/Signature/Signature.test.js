import React from 'react';
import { shallow } from 'enzyme';

import Signature from './Signature';

it('Renders', () => {
  const props = {
    email: 'your.name@readify.net',
    isSupport: false,
    mobile: '+61 111 222 333',
    name: 'your name',
    placeholders: {
      email: 'your.name@readify.net',
      mobile: '+61 111 222 333',
      name: 'Your Name',
      qualifications: 'Jedi Master | PSM I',
      title: 'Job Title',
      twitter: '@myTwitter',
    },
    qualifications: 'Jedi Master | PSM I',
    title: 'Job Title',
    twitter: '@myTwitter',
    brandLink: 'https://test.com',
    brandLinkName: 'Test',
    brandLogo: 'https://test.com/test-image'
  };

  const tree = shallow(<Signature {...props} />);
  expect(tree).toMatchSnapshot();
});