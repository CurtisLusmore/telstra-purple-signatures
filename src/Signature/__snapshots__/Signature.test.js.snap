// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div>
  <table
    border="0"
    cellPadding="0"
    cellSpacing="0"
    style={
      Object {
        "color": "#3D5567",
        "fontFamily": "Segoe UI, Tahoma, Geneva, Verdana, sans-serif",
        "fontSize": "10.5pt",
      }
    }
    width="100%"
  >
    <tr>
      <td
        style={
          Object {
            "borderRight": "1px solid #3D5567",
            "paddingRight": "20px",
            "width": "76px",
          }
        }
        valign="top"
      >
        <img
          height="53"
          src={[Function]}
          width="73"
        />
      </td>
      <td
        style={
          Object {
            "paddingLeft": "20px",
          }
        }
        valign="top"
      >
        <p
          style={
            Object {
              "marginBottom": "10px",
            }
          }
        >
          <b>
            your name
          </b>
          <br />
           | 
          Job Title
        </p>
        <p
          style={
            Object {
              "marginBottom": "10px",
            }
          }
        >
          Jedi Master | PSM I
        </p>
        <p
          style={
            Object {
              "marginBottom": "20px",
            }
          }
        >
          <span>
            <b>
              M
            </b>
             
             
          </span>
          <span>
              
          </span>
          <span>
            <b>
              E
            </b>
             
             
          </span>
          <span>
            <b>
              W
            </b>
             
            <a
              href="https://test.com"
            >
              Test
            </a>
          </span>
        </p>
        <SocialMedia />
      </td>
    </tr>
  </table>
</div>
`;
